cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -596,25 +600,28 @@ public void markApplicationsLatest(Collection<ApplicationId> appIds)
   @Override
   public int addLatestApplication(ApplicationId id, ApplicationMeta meta) throws ConflictException {
     return TransactionRunners.run(transactionRunner, context -> {
+      getSourceControlMetadataStore(context).setNamespaceSourceControlMeta(id, meta.getSourceControlMeta());
       return getAppMetadataStore(context).createLatestApplicationVersion(id, meta);
     }, ConflictException.class);
   }
 
   @Override
   public int addApplication(ApplicationId id, ApplicationMeta meta, boolean isLatest) throws ConflictException {
     return TransactionRunners.run(transactionRunner, context -> {
+      getSourceControlMetadataStore(context).setNamespaceSourceControlMeta(id, meta.getSourceControlMeta());
       return getAppMetadataStore(context).createApplicationVersion(id, meta, isLatest);
     }, ConflictException.class);
   }
 
   @Override
-  public void updateApplicationSourceControlMeta(Map<ApplicationId, SourceControlMeta> updateRequests)
+  public void updateApplicationSourceControlMeta(
+      Map<ApplicationId, SourceControlMeta> updateRequests)
       throws IOException {
     TransactionRunners.run(transactionRunner, context -> {
-      AppMetadataStore mds = getAppMetadataStore(context);
+      SourceControlMetadataStore mds = getSourceControlMetadataStore(context);
       for (Map.Entry<ApplicationId, SourceControlMeta> updateRequest : updateRequests.entrySet()) {
         try {
-          mds.updateAppScmMeta(updateRequest.getKey(), updateRequest.getValue());
+          mds.updateNamespaceSourceControlMeta(updateRequest.getKey(), updateRequest.getValue());
```

Comment:
previously, this would be a no-op if the application did not exist. That behavior needs to be maintained (for example, by checking for the app with AppMetadataStore), otherwise it is possible for the tables to get out of sync. For example, if this method is called for deleted applications, the application will exist in the source control table but not in the applications table.

Reply:
Noted. I have made the required change of checking for app existence with AppMetadataStore using getLatest.